src/EditorFeatures/TestUtilities2/Intellisense/TestStateFactory.vb

```diff
@@ -32,17 +38,19 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
                                                             workspaceElement As XElement,
                                                             Optional extraCompletionProviders As CompletionProvider() = Nothing,
                                                             Optional extraExportedTypes As List(Of Type) = Nothing,
-                                                            Optional workspaceKind As String = Nothing) As TestState
-
-            If completionImplementation = CompletionImplementation.Legacy Then
-                Return TestState.CreateTestStateFromWorkspace(workspaceElement, extraCompletionProviders, extraExportedTypes, workspaceKind)
-            End If
+                                                            Optional workspaceKind As String = Nothing) As TestStateBase
+            Select Case completionImplementation
+                Case CompletionImplementation.Legacy
+                    Return TestState.CreateTestStateFromWorkspace(workspaceElement, extraCompletionProviders, extraExportedTypes, workspaceKind)
+                Case CompletionImplementation.Modern
+                    Return ModernCompletionTestState.CreateTestStateFromWorkspace(workspaceElement, extraCompletionProviders, extraExportedTypes, workspaceKind)
+            End Select
 
             Throw New ArgumentException(completionImplementation.ToString())
         End Function
 
         Public Shared Function GetAllCompletionImplementations() As IEnumerable(Of Object())
```

Comment:
weird to me that this is an array of arrays...

Reply:
Each InlineData for xUnit theory is an array of objects. If a test needs to support more than 1 InlineData, we need to provide an array of InlineData, i.e. an array of arrays.

---
In reply to: [232508185](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508185) [](ancestors = 232508185)